modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/_diffs/plugins/uibutton/plugin.js

```diff
@@ -30,6 +30,7 @@
 		' title="{title}"' +
 		'>' +
 		'<span class="cke_button_icon cke_button__{icon}_icon">&nbsp;</span>' +
+		'<span class="cke_button_label cke_button__{label}_label" style="display:inline-block;">{label}</span>' +
```

Comment:
I'm on it

Reply:
When using `new CKEDITOR.template` as you mentioned it shows:

<img width="912" alt="Screen Shot 2021-11-05 at 18 12 34" src="https://user-images.githubusercontent.com/7663513/140579218-fbdde373-0d81-491d-8828-d6bf22ed0879.png">

It is caused because we're needing [a string ](https://github.com/ckeditor/ckeditor4/blob/4e64f672190a14d3c38aebd94ae4fb10c79e4545/core/ckeditor.js#L87)for registering the template of `balloonToolbarButton` and when using template instance we have an object. Inside this object it's just registered a function called source containing the callback we provided to create our conditionals.

I tried to use this instance directly [instead of template.output](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/_diffs/plugins/uibutton/plugin.js#L89) but it still not being shown.

I searched around ckeditor4 repo and I can only see [long string concats with conditionals](https://github.com/ckeditor/ckeditor4/blob/62e3ae4c0dd1ddb37882ca0cfcf8078a23b986b1/plugins/richcombo/plugin.js#L15..L55) 😂

Reply:
@diegonvs If you prefer to use `addTemplate` utility, that's fine. It is using [new CKEDITOR.template](https://github.com/ckeditor/ckeditor4/blob/4e64f672190a14d3c38aebd94ae4fb10c79e4545/core/ckeditor.js#L99) under the hood anyway. I am not sure if firing `'template'` event does anything in out case, probably not.

I was referring to conditional template string building over buffer `output` array. That's better than string concats you linked :slightly_smiling_face:

Reply:
> I am not sure if firing 'template' event does anything in our case, probably not.

Yes, it needs to be research

> I was referring to conditional template string building over buffer output array. That's better than string concats you linked 

Yes, I know but We can't do it because building the template using output requires some arguments like [`clickFn` ](https://github.com/liferay-frontend/liferay-portal/blob/24dfbcbb40dab5bc9b1decf46a06a0836f3c7f39/modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/_diffs/plugins/uibutton/plugin.js#L83) that are not reachable at this part of the code

Reply:
but revisithing this problem now, after two weeks I saw we can move the template definition to inside render method before using it. I'll do that.

Reply:
Done!